compiler/cli/cli-common/src/org/jetbrains/kotlin/cli/common/Properties.kt

```diff
@@ -17,3 +17,11 @@
 package org.jetbrains.kotlin.cli.common
 
 public val KOTLIN_COMPILER_ENVIRONMENT_KEEPALIVE_PROPERTY = "kotlin.environment.keepalive"
+
+
+fun String?.toBooleanLenient(): Boolean? = when (this?.toLowerCase()) {
```

Comment:
It seems it is possible to simplify conditional expressions by using the following function instead of `toBooleanLenient`
```
fun String.systemPropertyAsBooleanOrTrueOtherwise(negate: Boolean): Boolean {
    val propValue = System.getProperty(this) ?: return true

    return when (propValue.toLowerCase()) {
        in listOf("", "yes", "true", "on", "y") -> !negate
        in listOf("no", "false", "off", "n") -> negate
        else -> true
    }
}
```
See comments below.